python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1155,6 +1209,19 @@ def submit_run_request(
     instance.submit_run(run.run_id, workspace)
 
 
+def _execution_plan_targets_asset_selection(
+    execution_plan_snapshot: ExecutionPlanSnapshot, asset_selection: Sequence[AssetKey]
+) -> bool:
+    output_asset_keys = set()
+    for step in execution_plan_snapshot.steps:
+        if step.key in execution_plan_snapshot.step_keys_to_execute:
+            for output in step.outputs:
+                asset_key = check.not_none(output.properties).asset_key
+                if asset_key:
+                    output_asset_keys.add(asset_key)
+    return all(key in output_asset_keys for key in asset_selection)
```

Comment:
in theory we could be stricter here right? (`output_asset_keys == set(asset_selection)`)